include/igl/opengl/ViewerData.cpp

```diff
@@ -260,6 +264,44 @@ IGL_INLINE void igl::opengl::ViewerData::set_texture(
   dirty |= MeshGL::DIRTY_TEXTURE;
 }
 
+IGL_INLINE void igl::opengl::ViewerData::set_data(
+  const double caxis_min,
+  const double caxis_max,
+  const Eigen::VectorXd & D)
+{
+  if(!show_texture)
+  {
+    Eigen::MatrixXd CM;
+    igl::parula(Eigen::VectorXd::LinSpaced(21,0,1).eval(),false,CM);
```

Comment:
We should add an option `igl::ColorMapType` to specify the colormap. Also, our colormaps are defined using 256 samples in `igl/colormap.cpp`, so we should use the same number of samples here.

Reply:
see other comment. 21 is to create identifiable intervals which helps with scalar field visualization.